src/components/Users/UserSelector.tsx

```diff
@@ -0,0 +1,75 @@
+import {
+  FC, useState, useEffect, useRef,
+} from 'react';
+import classNames from 'classnames';
+import { useUsersContext } from '../../hooks/useUsersContext';
+import { Loader } from '../Loader';
+import { UsersList } from './UsersList';
+
+export const UserSelector: FC = () => {
+  const { isUsersLoading, selectedUser } = useUsersContext();
+
+  const [dropdownIsActive, setDropdownIsActive] = useState(false);
+
+  const dropdownRef = useRef<HTMLDivElement | null>(null);
+
+  const toggleDropdown = () => {
+    setDropdownIsActive(prevState => !prevState);
+  };
+
+  const closeDropdown = () => setDropdownIsActive(false);
+
+  const handleClickOutsideDropdown = (event: MouseEvent) => {
+    const dropdown = dropdownRef.current;
+
+    if (dropdown && !dropdown.contains(event.target as Node)) {
+      closeDropdown();
+    }
+  };
+
+  useEffect(() => {
+    document.addEventListener('click', handleClickOutsideDropdown);
+
+    return () => {
+      document.removeEventListener('click', handleClickOutsideDropdown);
+    };
+  }, []);
```

Comment:
to avoid recreations of this `fn` on each rendering
```suggestion
  useEffect(() => {
    const handleClickOutsideDropdown = (event: MouseEvent) => {
      const dropdown = dropdownRef.current;
  
      if (dropdown && !dropdown.contains(event.target as Node)) {
        closeDropdown();
      }
    };

    document.addEventListener('click', handleClickOutsideDropdown);

    return () => {
      document.removeEventListener('click', handleClickOutsideDropdown);
    };
  }, []);
```